src/lib/y2firstboot/clients/user.rb

```diff
@@ -77,18 +80,21 @@ def run
       #
       # For example, the home directory is modified to keep it on sync with the user name.
       def update_user
-        home_path = Pathname.new(user.home || "")
+        user.home ||= Y2Users::Home.new("")
 
-        return if user.home.nil? || user.name == home_path.basename.to_s
+        home_path = Pathname.new(user.home.path || "")
 
-        user.home = home_path.dirname.join(user.name).to_s
+        return if user.home.path.nil? || user.name == home_path.basename.to_s
+
+        user.home.path = home_path.dirname.join(user.name).to_s
       end
 
       # Writes config to the system
       def write_config
         writer = Y2Users::Linux::Writer.new(
           config,
-          Y2Users::ConfigManager.instance.system
+          Y2Users::ConfigManager.instance.system,
+          Y2Users::CommitConfigCollection.new.tap { |collection| collection.add(commit_config) }
```

Comment:
NP: for one-liner blocks we use single letter for the params: ` { |c| c.add }`

Reply:
BTW: I would extract the generation of the `CommitConfigCollection` to a separate method:

~~~
def commit_configs
  CommitConfigCollection.new.tap do |collection|
    config = CommitConfig.new.tag { ... }
    collection.add(config)
  end
end
~~~